flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/casting/MapAndMultisetToStringCastRule.java

```diff
@@ -109,16 +104,14 @@ protected String generateCodeBlockInternal(
             String returnVariable,
             LogicalType inputLogicalType,
             LogicalType targetLogicalType) {
-        boolean isMap = inputLogicalType instanceof MapType;
-        final LogicalType keyType;
-        final LogicalType valueType;
-        if (isMap) {
-            keyType = ((MapType) inputLogicalType).getKeyType();
-            valueType = ((MapType) inputLogicalType).getValueType();
-        } else {
-            keyType = ((MultisetType) inputLogicalType).getElementType();
-            valueType = INT().getLogicalType();
-        }
+        final LogicalType keyType =
+                inputLogicalType.is(LogicalTypeFamily.COLLECTION)
```

Comment:
Shouldn't this be `inputLogicalType.is(LogicalTypeRoot.MULTISET)`?

Reply:
makes sense, probably I didn't get your previous comment as you referred to LogicalTypeFamily

Reply:
I mispelled it, sorry :smile:

Reply:
ok, done, thanks